### DISCUSSION TITLE
Categorize feature proposals with labels

### DISCUSSION THREAD

**Discussion by gulshan** [dotnet/csharplang#726] [General]
I think there are already a lot of issues containing new C# feature proposals in this repo. And categorization/classification of these issues with appropriate labels can bring some benefits like easy exploration and discussions. The categorization can be done from different perspective also. For example, the first thing comes to my mind is the proposals that enables the language to do things not possible now vs. proposal for syntactic sugars. The issues regarding type systems can have a separate label. The pattern matching, generic programming or functional constructs can have their own labels and so on. This can also increase community engagement. Their can be labels specifying needs for more specific design, grimmer or prototype implementations. Then there can be introduction of some metrics for quantification of merits of a proposal. Overall, this repo and C# language design process can get more structured and open IMHO.

**Comment by scottdorman** [comment]
The question of labels (and what they should be and that they should be consistent between both the C# and VB language repos) was brought up back in February as well (https://github.com/dotnet/csharplang/issues/54) but the discussion didn't really go anywhere.

**Comment by Thaina** [comment]
This repo should have some label tagger bot that would read some syntax to work

Such as, we could have agreement that if the last line of any issue started with `tags:a,b,c,d` The bot would add those a,b,c,d tags

**Comment by Unknown6656** [comment]
@Thaina : that could possibly go wrong, if one adds the wrong tags to a question/issue
One should restrict the user-added tags, otherwise people could add the tag `[Proposal champion]` very easily.

**Comment by Thaina** [comment]
@Unknown6656 Well it sure need some constraint. Actually it must be a whitelist of available tags only for what it is such as `question`,`request`,`bug` or which part it requested. such as `generic`,`interface`,`operator` etc. but not an admin tag like that.

Maybe limit only 5 tag
